### PR TITLE
Enable distributed 3D simulation option

### PR DESCRIPTION
**Pull request by akshaysridhar**
Modify function spaces to enable distributed simulations
cc @sriharshakandala 

- [x] Distributed 3D Box  
- [x] Distributed 3D Sphere
- [ ] Replicate ClimaCore tests with modifications

### CODE REVIEW

**Comment by akshaysridhar** [comment]
Awaiting changes from ClimaCore #631 before merging this (it should simplify ClimaAtmos changes a fair bit)

File: src/Simulations/simulation.jl [hunk 1]
```diff
@@ -28,8 +28,23 @@ function Simulation(
...
+        logger_stream = ClimaComms.iamroot(Context) ? stderr : devnull
+        prev_logger = global_logger(ConsoleLogger(logger_stream, Logging.Info))
+        atexit() do
+            global_logger(prev_logger)
+        end
+    else
+        global_logger(TerminalLogger())
+    end
+
+    space_center, space_face, comms_ctx_center, comms_ctx_face =
```
Comment (bischtob) [inline src/Simulations/simulation.jl:46]: `comms_ctx_xyz` could be `comms_context_xyz`

File: test/test_domains.jl [hunk 2]
```diff
@@ -27,10 +27,6 @@ float_types = (Float32, Float64)
         @test domain.nelements == 2
         @test domain.stretching isa Meshes.StretchingRule
         @test domain.stretching isa Uniform
```
Comment (bischtob) [inline test/test_domains.jl:29]: add a todo for the upcoming tests

**Review by bischtob — CHANGES_REQUESTED** [review]
Nice. Just a few tiny things.

File: src/Models/Nonhydrostatic3DModels/equations_base_model.jl [hunk 3]
```diff
@@ -135,10 +135,10 @@ function rhs_base_model!(
         @. dρ += flux_correction_center(w, ρ)
     end
 
-    # discrete stiffness summation for spectral operations
-    Spaces.weighted_dss!(dρ)
-    Spaces.weighted_dss!(duh)
-    Spaces.weighted_dss!(dw)
+    # discrete stiffness summation for spectral operations, distributed
+    Spaces.weighted_dss!(dρ, comms_ctx_center)
```
Comment (sriharshakandala) [inline src/Models/Nonhydrostatic3DModels/equations_base_model.jl:139]: We are currently using a conservatively sized comms_ctx for both face and center DSS in ClimaCore

**Comment by akshaysridhar** [comment]
631 in ClimaCore is in, I'll udpate this PR  and request a second review.

**Pull request closed without merging**